crypto/src/vrf/schnorrkel/mod.rs

```diff
@@ -0,0 +1,158 @@
+use crate::random::{CryptoRng, Rng};
+use parity_scale_codec::{Decode, Encode};
+
+const EXPECTED_PUBKEY_LEN: usize = 32;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[must_use]
+pub struct SchnorrkelPublicKey {
+    key: schnorrkel::PublicKey,
+}
+
+impl Encode for SchnorrkelPublicKey {
+    fn encode(&self) -> Vec<u8> {
+        self.key.to_bytes().to_vec()
+    }
```

Comment:
It's probably more appropriate to implement `using_encoded` instead of `encode` for types that can be cheaply converted to `&[u8]` encoded representation. That avoids allocating an extra `Vec` in many cases and you get efficient `encoded_size` for free.

Reply:
Alrighty